### PR TITLE
journalctl-authenticate: several cleanups

### PR DESCRIPTION
**Pull request by yuwata**
Addresses https://github.com/systemd/systemd/pull/31879#discussion_r1539063239.

### CODE REVIEW

File: src/journal/journalctl-authenticate.c [hunk 1]
```diff
@@ -155,10 +149,10 @@ int action_setup_keys(void) {
         if (r < 0)
                 return log_error_errno(r, "Failed to write state: %m");
 
-        if (rename(k, p) < 0)
+        if (rename(tmpfile, path) < 0)
```
Comment (YHNdnzj) [inline src/journal/journalctl-authenticate.c]: `open_tmpfile_linkable` should be used with `link_tmpfile`?

File: src/journal/journalctl-authenticate.c [hunk 2]
```diff
@@ -87,21 +76,16 @@ int action_setup_keys(void) {
...
+        if (!path)
                 return log_oom();
 
         if (arg_force) {
-                r = unlink(p);
-                if (r < 0 && errno != ENOENT)
-                        return log_error_errno(errno, "unlink(\"%s\") failed: %m", p);
-        } else if (access(p, F_OK) >= 0)
+                if (unlink(path) < 0 && errno != ENOENT)
+                        return log_error_errno(errno, "unlink(\"%s\") failed: %m", path);
```
Comment (YHNdnzj) [inline src/journal/journalctl-authenticate.c]: Also modernize the log message while at it? "Failed to ..."?

File: src/journal/journalctl-authenticate.c [hunk 3]
```diff
@@ -157,61 +139,65 @@ int action_setup_keys(void) {
...
+                FORMAT_TIMESPAN(arg_interval, 0),
+                ansi_highlight(), ansi_normal(),
+                ansi_highlight_red());
+        fflush(stderr);
+
         puts(key);
 
-        if (on_tty()) {
-                fprintf(stderr, "%s", ansi_normal());
+        fprintf(stderr, "%s", ansi_normal());
```
Comment (YHNdnzj) [inline src/journal/journalctl-authenticate.c]: fputs?

**Review by YHNdnzj — APPROVED** [review]

**Comment by yuwata** [comment]
@YHNdnzj Thank you for the review. The above suggestions are applied. Upgrading the green label.

**Pull request merged**